mlbench_core/dataset/imagerecognition/pytorch/dataloader.py

```diff
@@ -91,25 +90,3 @@ def __init__(self, root, train=True):
             self.root = os.path.join(self.root, 'val')
 
         super().__init__(self.root, transform)
-
-
-def partition_dataset_by_rank(dataset, rank, world_size, distribution='uniform', shuffle=True):
```

Comment:
This function is used many places in `mlbench-benchmarks` (e.g. in https://github.com/mlbench/mlbench-benchmarks/blob/896d3c4f31aa7205ad9fcedebd5dc46e3b8e36ed/pytorch/imagerecognition/openmpi-cifar10-resnet20-all-reduce/main.py#L78). Maybe this needs to be changed too before merging

Reply:
Yes definitely, at the moment is copied in 2-3 places already which is why i want to move it to a single file. Though unfortunately it doesn't work with torchtext datasets at the moment.